### ISSUE TITLE
RTL support appears to be incomplete

### ISSUE THREAD

**Issue by amire80** [WhisperSystems/Signal-Android#4714]
Hi,

So I just received an update of Signal with complete translation to Hebrew, and it's WONDERFUL. Thank you for having such a great localization process - not all apps are so smooth.

However, since Hebrew is written right-to-left, as are Arabic, Persian and a few other languages, it requires a bit of extra work to make it properly functional in RTL.

For example, in the Settings menu all the items are aligned to the left, and they should be aligned to the right.

I'm not an Android developer, but as far as I know, since Android 4 or so, most of this can be done by enabling auto-support for RTL somewhere in the app's build definitions (I'm sorry if my language here is imprecise).

**Comment by mcginty** [comment]
Yeah, we still have a lot of hardcoded "left" and "right" values instead of the more modern "start" and "end" to support RTL more smoothly as well.

**Comment by amire80** [comment]
Screenshot:

![screenshot_2015-11-25-22-09-31](https://cloud.githubusercontent.com/assets/346271/11408567/b461226c-93c2-11e5-9063-49d7de9aa402.png)

**Comment by amire80** [comment]
In the screenshot:
* the icons are supposed to be on right
* their labels are supposed to be right-aligned
* the title at the top (Settings) is supposed to be right-aligned
* the arrow near it is supposed to point to the right.

**Comment by agrajaghh** [comment]
I'll have a look at this

**Comment by agrajaghh** [comment]
@amire80 could you help me testing and try this branch: https://github.com/agrajaghh/Signal-Android/tree/RTL_fix ? I know the incoming messages are missing, but what else needs to be fixed?

**Comment by agrajaghh** [comment]
incoming messages are fixed now. @amire80 Do I need to swap everything between left and right? At the moment incoming messages are on the left and outgoing on the right. Is this still okay for RTL languages? Swapping them might not be easy because of the bubble design...

**Comment by amire80** [comment]
Heh, the messages' alignment is a thing to which I didn't give that much thought till you asked!

So I now I did a little comparison and here's what I found: When the phone's UI is Hebrew, the default SMS app and Google Hangouts put the outgoing messages on the right and the incoming messages on the left. Telegram does the opposite. However, when the phone's UI is English all the apps put the outgoing messages on the left and the incoming messages on the right.

So I guess that Telegram doesn't flip it for Hebrew, but Signal probably should flip them.

Summary: the outgoing messages should be on the right and the incoming messages should be on the left.

**Comment by amire80** [comment]
As for you previous comment, if you suggest that I build an app from your source, then I cannot do this, unfortunately, because I cannot install the Android developer tools on my laptop at the moment (I hope to do that in a couple of months).

If you can post screenshots of the settings menu and a sample of messages, I can say whether it looks correct.

Thank you very much for working on this!

**Comment by moxie0** [comment]
@agrajaghh fyi I don't think we'd accept a change that flips the message bubbles, too complicated for too little signficiance

**Comment by amire80** [comment]
Yeah, flipping the message bubbles is at most nice-to-have. I used Telegram, Hangout and SMS without even noticing the differences till now.

**Comment by agrajaghh** [comment]
@moxie0 
> @agrajaghh fyi I don't think we'd accept a change that flips the message bubbles, too complicated for too little signficiance

yep, I thought so. What about changing the ConversationList? At the moment the avatar is on the left and the date/time on the right. Would you be okay with swapping them?
